src/main/java/ShuntingYardCalculator/OptionsFactory.java

```diff
@@ -0,0 +1,20 @@
+package ShuntingYardCalculator;
+
+import ShuntingYardCalculator.Enums.MenuOption;
+
+import static ShuntingYardCalculator.Enums.MenuOption.*;
+
+public class OptionsFactory {
+    public static MenuOption factory(String chosenOption) {
+        switch (chosenOption) {
+            case "1":
```

Comment:
you can pass a string to your choice to the enum, It does not have to be in range from 1 to infinity